src/SSVOpenHexagon/Core/HGUpdate.cpp

```diff
@@ -25,14 +25,28 @@ void HexagonGame::update(ssvu::FT mFT)
 {
     mFT *= Config::getTimescale();
 
-    // TODO: refactor to avoid repetition, and truncate floating point number
-    // TODO: also show best record (here) and last run + best record (in menu)
-    steamManager.set_rich_presence_in_game(
-        levelData->name, status.getTimeSeconds());
+    // TODO: show best record (here) and last run + best record (in menu)
+
+    // Name formatter
+    std::string nameStr = std::string(levelData->name);
+    nameStr[0] = std::toupper(nameStr[0]);
+
+    // Difficulty multipler formatter
+    std::string diffStr = std::to_string(HexagonGame::difficultyMult);
+    size_t endPos = diffStr.find_last_not_of('0');
+    diffStr.erase(endPos + 1 + (int)(diffStr[endPos] == '.'), std::string::npos); // at least 1 dp
+
+    // Time formatter
+    std::string timeStr = std::to_string(floorf(status.getTimeSeconds() * 1000) / 1000); // 3 dp
+    timeStr.erase(timeStr.find_first_of('.') + 4, std::string::npos);
```

Comment:
Again, this is not a convenient way of formatting... if you're trying to limit the number of floating point decimals, why not use something like `setprecision` or good ol'`printf`? https://stackoverflow.com/a/14432132/598696